instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/jmx/AttributeValueExtractor.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.runtimemetrics.jmx;
+
+import static java.util.logging.Level.FINE;
+import static java.util.logging.Level.INFO;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanAttributeInfo;
+import javax.management.MBeanInfo;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import javax.management.openmbean.CompositeData;
+import javax.management.openmbean.TabularData;
+
+/**
+ * A class responsible for extracting attribute values from MBeans. Objects of this class are
+ * immutable.
+ */
+public class AttributeValueExtractor implements LabelExtractor {
+  // The attribute name to be used during value extraction from MBean
+  private final String baseName;
+
+  // In case when the extracted attribute is a CompositeData value,
+  // how to proceed to arrive at a usable elementary value
+  private final String[] nameChain;
+
+  /**
+   * Verify the attribute name and create a corresponding extractor object.
+   *
+   * @param rawName the attribute name, can be a reference to composite values
+   * @return the corresponding LabelValueExtractor
+   * @throws IllegalArgumentException if the attribute name is malformed
+   */
+  public static AttributeValueExtractor fromName(String rawName) {
+    // Check if a CompositeType value is expected
+    int k = rawName.indexOf('.');
+    if (k < 0) {
+      if (rawName.isEmpty()) {
```

Comment:
We should check this condition before we look for the presence of `.`.

Reply:
The code does look fanny, but it is not necessary to check for emptiness if the string contains a dot. If code readability is affected, I can reverse the order or add a clarifying comment. Please let me know what you prefer.

Reply:
Similarly, its not necessary to check if the string contains a dot if the string is empty 🙂. Less nested conditions is generally better.